### PR TITLE
Zombie TODOs 0.12 pass

### LINKED ISSUE #4498: Rerun 0.12 zombie TODO pass
- [x] ./crates/re_viewer/src/ui/welcome_screen/welcome_page.rs+14: const CPP_SPAWN_MARKDOWN: &str = include_str!("../../../data/quick_start_guides/cpp_connect.md"); // TODO(#3757)
- [x] ./crates/re_viewer/src/ui/welcome_screen/welcome_page.rs+28: include_str!("../../../data/quick_start_guides/quick_start_connect.cpp"); // TODO(#3757)
- [x] ./crates/re_arrow_store/src/store_write.rs+228: // TODO(#1777): shared slices for auto generated keys
- [x] ./crates/re_data_store/src/entity_properties.rs+80: // TODO(#1423): We need to properly split entity properties that only apply to specific
- [x] ./rerun_cpp/src/rerun/collection.hpp+30: ///         TODO(#3794): don't do std::vector
- [x] ./rerun_cpp/src/rerun/collection.hpp+277: /// TODO(#4225): So far it always is continuous, but in the future we want to support strides!
- [x] ./rerun_cpp/src/rerun/archetypes/boxes3d.hpp+118: /// TODO(#3794): This should not take an std::vector.
- [x] ./rerun_cpp/src/rerun/archetypes/boxes3d.hpp+126: /// TODO(#3794): This should not take an std::vector.
- [x] ./rerun_cpp/src/rerun/archetypes/boxes3d.hpp+140: /// TODO(#3794): This should not take an std::vector.
- [x] ./rerun_cpp/src/rerun/archetypes/boxes3d_ext.cpp+34: /// TODO(#3794): This should not take an std::vector.
- [x] ./rerun_cpp/src/rerun/archetypes/boxes3d_ext.cpp+42: /// TODO(#3794): This should not take an std::vector.
- [x] ./rerun_cpp/src/rerun/archetypes/boxes3d_ext.cpp+56: /// TODO(#3794): This should not take an std::vector.
- [x] ./rerun_cpp/docs/readme_snippets.cpp+14: // TODO(#3794): Once image logging is nicer, we should do that in this snippet as well!
- [x] ./rerun_cpp/docs/writing_docs.md+14: TODO(#3974): get this part working for C++!
- [x] ./rerun_cpp/tests/recording_stream.cpp+131: // TODO(#2889): Setting thread locals causes a crash on shutdown on macOS.
- [x] ./docs/code-examples/roundtrips.py+29: "quick_start_spawn": ["cpp"], # TODO(#3870): Not yet implemented in C++
- [x] ./docs/content/reference/viewer/blueprint.md+43: [TODO(#1174)](https://github.com/rerun-io/rerun/issues/1174): Editable & user defined Data Blueprint Groups
- [x] ./tests/python/gc_stress/many_large_single_row_recordings.py+30: # TODO(#4183): Investigate this.

### PR DESCRIPTION
**Pull request by teh-cmc**
Title.

- Fixes #4498 

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/4633/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/4633/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/4633/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4633)
- [Docs preview](https://rerun.io/preview/75d5b94ac554a1926a390c4e3047f296002cdd7d/docs) 
- [Examples preview](https://rerun.io/preview/75d5b94ac554a1926a390c4e3047f296002cdd7d/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: docs/content/reference/viewer/blueprint.md [hunk 1]
```diff
@@ -41,8 +41,6 @@ Groups, despite being derived from the [Entity Path](../../concepts/entity-path.
 They are meant to improve the handling of large Space Views and allow for hierarchical manipulation
 of blueprints.
 
-[TODO(#1174)](https://github.com/rerun-io/rerun/issues/1174): Editable & user defined Data Blueprint Groups
```
Comment (teh-cmc) [inline docs/content/reference/viewer/blueprint.md:44]: This entire corner of the docs will likely be rewritten with the blueprint stuff shipping soon.

File: rerun_cpp/docs/readme_snippets.cpp [hunk 2]
```diff
@@ -12,8 +12,6 @@ static std::vector<rerun::Color> create_colors() {
     return {};
 }
 
-// TODO(#3794): Once image logging is nicer, we should do that in this snippet as well!
```
Comment (teh-cmc) [inline rerun_cpp/docs/readme_snippets.cpp:15]: Not sure what this was referring to, there are no images in there :shrug:

File: rerun_cpp/src/rerun/archetypes/boxes3d_ext.cpp [hunk 3]
```diff
@@ -32,15 +32,15 @@ namespace rerun {
         ///
         /// TODO(#3285): Does *not* preserve data as-is and instead creates half-sizes from the
         /// input data.
-        /// TODO(#3794): This should not take an std::vector.
+        /// TODO(andreas): This should not take an std::vector.
```
Comment (teh-cmc) [inline rerun_cpp/src/rerun/archetypes/boxes3d_ext.cpp:35]: 1. This not just this file but all of the extensions.
2. I don't want/don't have time to be playing with fire right now :grimacing:

File: rerun_cpp/docs/writing_docs.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ pixi run cpp-docs
 They then can be locally viewed at `rerun_cpp/docs/html/index.html`
 
 ### How versioned docs are generated and served
-TODO(#3974): get this part working for C++!
+TODO(andreas): get this part working for C++!
```
Comment (teh-cmc) [inline rerun_cpp/docs/writing_docs.md:15]: I don't have any clue myself

File: rerun_cpp/src/rerun/archetypes/boxes3d_ext.cpp (hunk 3 above)
Comment (emilk) [reply rerun_cpp/src/rerun/archetypes/boxes3d_ext.cpp:35]: @Wumpf

File: rerun_cpp/docs/writing_docs.md (hunk 4 above)
Comment (emilk) [reply rerun_cpp/docs/writing_docs.md:15]: @Wumpf

**Review by emilk — APPROVED** [review]
Thanks for cleaning this up! Looks like @Wumpf should also take a look

File: rerun_cpp/src/rerun/archetypes/boxes3d_ext.cpp (hunk 3 above)
Comment (Wumpf) [reply rerun_cpp/src/rerun/archetypes/boxes3d_ext.cpp:35]: let's leave it be. It affects a handful of extensions but not _that_ many. (e.g. just a few lines above things are `Collection` as they should be)

File: rerun_cpp/docs/writing_docs.md (hunk 4 above)
Comment (Wumpf) [reply rerun_cpp/docs/writing_docs.md]: I filled it out!

File: rerun_cpp/docs/readme_snippets.cpp (hunk 2 above)
Comment (Wumpf) [reply rerun_cpp/docs/readme_snippets.cpp:15]: yeah that was the problem, at the time of writing we wanted to put images in there but they were awful to log. Now it's nice. Commited that in :)

**Review by Wumpf — APPROVED** [review]

**Pull request merged**